python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -528,100 +562,275 @@ def asset_key_for_input(self, input_name: str) -> AssetKey:
         else:
             return key
 
+    ### Partition methods
+
     @public
-    def asset_partition_key_for_output(self, output_name: str = "result") -> str:
-        """Returns the asset partition key for the given output. Defaults to "result", which is the
-        name of the default output.
+    def partition_key_for_asset(
```

Comment:
should these for_asset methods have `asset` be optional and work in the single asset case?

Reply:
yeah that seems reasonable